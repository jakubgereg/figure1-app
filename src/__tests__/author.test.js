import React from "react";
import ReactDOM from "react-dom";
import Author from "../components/author";
import { shallow } from "enzyme";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Author />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders username correctly", () => {
  const wrapper = shallow(<Author username={"Jakub"} />);
  const welcome = <span className="user-name">Jakub</span>;
  expect(wrapper.contains(welcome)).toEqual(true);
});

it("renders a snapshot", () => {
  const wrapper = shallow(<Author />);
  expect(wrapper).toMatchSnapshot();
});
