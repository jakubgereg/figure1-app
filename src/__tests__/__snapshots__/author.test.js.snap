// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Author />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="undefined profile"
          className="profile"
        />,
        <span
          className="user-name"
        />,
        undefined,
      ],
      "className": "author",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "undefined profile",
          "className": "profile",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "user-name",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="undefined profile"
            className="profile"
          />,
          <span
            className="user-name"
          />,
          undefined,
        ],
        "className": "author",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "undefined profile",
            "className": "profile",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "user-name",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
